core/renderers/measurables/inputs.js

```diff
@@ -142,7 +144,7 @@ goog.inherits(Blockly.blockRendering.StatementInput,
 Blockly.blockRendering.ExternalValueInput = function(input) {
   Blockly.blockRendering.ExternalValueInput.superClass_.constructor.call(this,
       input);
-  this.type = 'external value input';
+  this.type |= Blockly.blockRendering.Types.EXTERNAL_VALUE_INPUT;
```

Comment:
What's the rule for deciding whether to use `|=` or just `=` to set the type?

Reply:
It should always be |= since the base is 0 and it's okay to bitwise OR zero.

Reply:
It should only be an equal in the two base classes: 
- Measurable
- Row